pkg/cluster/denyassignment.go

```diff
@@ -17,8 +17,8 @@ func (m *manager) createOrUpdateDenyAssignment(ctx context.Context) error {
 	}
 
 	// needed for AdminUpdate so it would not block other steps
-	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == "" {
-		m.log.Print("skipping createOrUpdateDenyAssignment: SPObjectID is empty")
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
```

Comment:
Is this correct, sorry if I am missing any other discussion.
IMO, the deny assignment will be only skipped whenever the case is similar to:-
!m.doc.OpenShiftCluster.UsesWorkloadIdentity() && `m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == ""`

Reply:
:facepalm: You're right, thanks for catching this